### PR TITLE
Enemies ai

### PR DESCRIPTION
**Pull request by gitesthuman**
# Pull request review checklist 📝 ✅

This is a checklist that helps conduct a pull request review properly. These points are required to merge the PR

- [x] There are no merge conflicts

- [x] The pull request is named accordingly (description of the functionality - it
   will be visible in the changelog later)
   
- [x] Every task requirement (from Jira) is done

- [x] The app runs, it boots up properly (e.g. no syntax errors)

- [x] The proposed functionality is working properly (you need to pull this branch
   and check for yourself)

- [x] Each function, class or functionality is properly documented
   (short description of what it does, parameters or return values if they need
   any explanation)

### CODE REVIEW

**Review by gitesthuman — COMMENTED** [review]
amazing job, my guy! 🙌

File: dnd-bot/dnd_bot/logic/prototype/creature.py [hunk 1]
```diff
@@ -32,11 +35,317 @@ def __init__(self, x=0, y=0, sprite: str = '', name: str = 'Creature', hp: int =
         self.drops = drops
 
         # TODO set ai function depending on ai argument
```
Comment (Cellmer) [inline dnd-bot/dnd_bot/logic/prototype/creature.py:37]:
```suggestion
```

File: dnd-bot/dnd_bot/logic/prototype/creature.py [hunk 2]
```diff
@@ -32,11 +35,317 @@ def __init__(self, x=0, y=0, sprite: str = '', name: str = 'Creature', hp: int =
...
+        matrix[to_y][to_x] = 1
+
+        grid = Grid(matrix=matrix)
+        start = grid.node(from_x, from_y)
+        end = grid.node(to_x, to_y)
+
+        return AStarFinder().find_path(start, end, grid)[0]
+
+    def prepare_move_queue(self):
+        """prepare chain of creature's actions depending on current game state"""
```
Comment (Cellmer) [inline dnd-bot/dnd_bot/logic/prototype/creature.py:116]: trzeba udokumentować oznaczenia tej kolejki, co oznaczają 'M', 'A' itd.

**Review by Cellmer — COMMENTED** [review]
Trzeba trochę lepiej udokumentować kolejki ruchów tak żeby od razu było wiadomo o co chodzi jak ktoś spojrzy na funkcję, poza tym wszystko wydaje się w porządku.

**Review by Cellmer — APPROVED** [review]
Wygląda elegancko

**Comment by gitesthuman** [comment]
let's merge this big boy

**Pull request merged**